webapp/app/forms/steps/steuerlotse_step.py

```diff
@@ -150,14 +150,17 @@ def render(self, **kwargs):
         )
 
     @staticmethod
-    def _delete_dependent_data(data_field_identifier: list, stored_data: dict):
+    def _delete_dependent_data(stored_data: dict, pre_fixes:list=None, post_fixes:list=None):
         """This method filters the stored data. It deletes all the elements where the key includes the
         data_field_identifier. """
-        return dict(filter(
-            lambda elem: not any([elem[0].startswith(data_field_prefix) for data_field_prefix in data_field_identifier])
-                         and not any(
-                [elem[0].endswith(data_field_prefix) for data_field_prefix in data_field_identifier]),
-            stored_data.items()))
+        filtered_data = stored_data
```

Comment:
I can do the copy here; I did not do so because of performance . It is basically to ensure that we return something if neither prefix nor postfix are set.

Reply:
Yes, makes sense. just thinking that we could also just keep it as stored_data and modify that without possibly confusing with the re-assignment. But completely open to you, no strong opinion.

Reply:
Yes, but I would then feel not so good about it, because we do not want to modify the input but only the returned result here.